semgrep ci installs DeepSemgrep automatically when needed

IagoAbal opened the pull request:
Fixes: 44b45065750 ("semgrep ci runs DeepSemgrep when it is enabled in App (#6624)")

Reviewed-by: Brendon Go

test plan:
Remove deep-semgrep, enable DeepSemgrep in App/Settings, and run `semgrep ci` on a Java repo

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

brendongo commented on cli/src/semgrep/commands/ci.py:
```diff
@@ -323,6 +325,10 @@ def ci(
             # Run DeepSemgrep when available but only for full scans
             is_full_scan = metadata.merge_base_ref is None
             deep = scan_handler.deepsemgrep if scan_handler and is_full_scan else False
+            SemgrepCore.path()  # initializes a SemgrepCore class-field needed for .deep_path() to work
```
What class filed needs to be initialized?

IagoAbal replied:
Oh never mind, I just mixed things up it seems, it's not needed to call `SemgrepCore.path()` first.

IagoAbal committed:
semgrep ci installs DeepSemgrep automatically when needed

Fixes: 44b45065750 ("semgrep ci runs DeepSemgrep when it is enabled in App (#6624)")

Reviewed-by: Brendon Go

test plan:
Remove deep-semgrep, enable DeepSemgrep in App/Settings, and
run `semgrep ci` on a Java repo
